tests/conftest.py

```diff
@@ -5,7 +5,494 @@
+    """
+    Returns a constructor function for creating a Collection.
+    """
+    def _create_collection(db_session, name=None, protocol=ExternalIntegration.OPDS2_IMPORT,
+                           external_account_id=None, url=None, username=None,
+                           password=None, data_source_name=None):
+        name = name or "Good Reads"
+        collection, _ = get_one_or_create(db_session, Collection, name=name)
+        collection.external_account_id = external_account_id
+        integration = collection.create_external_integration(protocol)
+        integration.goal = ExternalIntegration.LICENSE_GOAL
+        integration.url = url
+        integration.username = username
+        integration.password = password
+
+        if data_source_name:
+            collection.data_source = data_source_name
+
+        return collection
+
+    return _create_collection
+
+
+@pytest.fixture
+def create_contributor():
+    """
+    Returns a constructor function for creating a Contributor.
+    """
+    def _create_contributor(db_session, sort_name=None, name=None, **kwargs):
+        name = sort_name or name
+        contributor, _ = get_one_or_create(db_session, Contributor, sort_name=name, **kwargs)
+
+        return contributor
+    
+    return _create_contributor
+
+
+@pytest.fixture
+def create_edition(create_licensepool):
+    """
+    Returns a constructor function for creating an Edition.
+    """
+    def _create_edition(db_session, data_source_name=DataSource.GUTENBERG, identifier_type=Identifier.GUTENBERG_ID,
+                        with_license_pool=False, with_open_access_download=False, title=None, language="eng",
+                        authors=None, identifier_id=None, series=None, collection=None, publication_date=None,
+                        self_hosted=False, unlimited_access=False):
+        id = identifier_id or random.randint(1, 9999)
+        id = str(id)
+        source = DataSource.lookup(db_session, data_source_name)
+        if not source:
+            # source can't be None, so for now this is the sensible default
+            source, _ = get_one_or_create(db_session, DataSource, name=data_source_name)
+        wr = Edition.for_foreign_id(db_session, source, identifier_type, id)[0]
+        if not title:
+            title = "Test Book"
+        wr.title = str(title)
+        wr.medium = Edition.BOOK_MEDIUM
+        if series:
+            wr.series = series
+        if language:
+            wr.language = language
+        if authors is None:
+            authors = "Test Author"
+        if isinstance(authors, str):
+            authors = [authors]
+        if authors:
+            primary_author_name = str(authors[0])
+            contributor = wr.add_contributor(primary_author_name, Contributor.PRIMARY_AUTHOR_ROLE)
+            if not contributor.display_name and ',' not in primary_author_name:
+                contributor.display_name = primary_author_name
+            wr.author = primary_author_name
```

Comment:
Stylistic thing that you don't need to agree with--I find it more readable to have an empty line to set off conditional blocks that aren't actually related or part of an if/elif/else construct. This kind of thing:

```Python
x = something()
if x:
    print("yup, x alright")

y = something()
if y:
    print("y is in fact true")

if x and y:
    print("both of them!")
elif x and not y:
    print("the x but not the y")
elif y and not x:
    print("the y but not the x")
else:
    print("neither of them")

return some_other_thing
```

I think it's more of a big deal in code that goes on longer than one vertical screen (~50 lines), because it gives it more visual texture as you scroll so it's easier to find your place quickly.